asm/macros/battle_script.inc

```diff
@@ -1713,6 +1713,12 @@
 	various \battler, VARIOUS_JUMP_IF_SHIELDS_DOWN_PROTECTED
 	.4byte \ptr
 	.endm
+
+	.macro handletypeimmunity battlerability:req, battlertypes:req, status:req
```

Comment:
I think `trypoisontype` would be a better name. Also would prefer if the battlescript pointer would be passed as an argument.
Would also get rid of the status argument and treat it only for trying to inflict Poison/Bad Poison

Reply:
My mindset was making it so it could support custom abilities similar to corrosion, but sure.

Reply:
Done.